nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -535,6 +566,44 @@ nav_msgs::msg::Path RegulatedPurePursuitController::transformGlobalPlan(
   return transformed_plan;
 }
 
+double RegulatedPurePursuitController::dot(double v1[], double v2[])
+{
+  // Supports only 2D
+  return  (v1[0]*v2[0]) + (v1[1]*v2[1]);
+}
+
+double RegulatedPurePursuitController::checkOrientation(const geometry_msgs::msg::PoseStamped & pose )
+{
+  double dot_sign = 0.0;
+  double iter = 1;
+  double cusp_distance = 0.0;
+  bool not_in_cusp = true;
+
+  do
```

Comment:
Restructure this as a for loop and actually use iterators, `iter` is an indexer and you can easily format this as a `for begin until end, iter++`. for loops are always preferable to while loops since you know the bounded maximum number iterations

Reply:
You can remove the entire `not_in_cusp` logic too since that is just there as a  terminal condition of the while loop, which the for would have just by exiting.

Reply:
Ah this is a cool idea! I will do change it! :)